v2/bundle/jwtbundle/bundle.go

```diff
@@ -1,73 +1,164 @@
 package jwtbundle
 
 import (
+	"bytes"
 	"crypto"
+	"encoding/json"
+	"errors"
+	"fmt"
 	"io"
+	"io/ioutil"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"gopkg.in/square/go-jose.v2"
 )
 
 // Bundle is a collection of trusted JWT public keys for a trust domain.
 type Bundle struct {
+	mtx            sync.RWMutex
+	trustDomain    spiffeid.TrustDomain
+	jwtSigningKeys map[string]crypto.PublicKey
 }
 
 // New creates a new bundle.
 func New(trustDomain spiffeid.TrustDomain) *Bundle {
-	panic("not implemented")
+	return &Bundle{
+		mtx:            sync.RWMutex{},
+		trustDomain:    trustDomain,
+		jwtSigningKeys: make(map[string]crypto.PublicKey),
+	}
 }
 
 // Load loads a Bundle from a file on disk.
 func Load(trustDomain spiffeid.TrustDomain, path string) (*Bundle, error) {
-	panic("not implemented")
+	bundleBytes, err := ioutil.ReadFile(path)
+
+	if err != nil {
+		return nil, fmt.Errorf("unble to read JWT bundle: %w", err)
+	}
+
+	return Parse(trustDomain, bundleBytes)
 }
 
 // Read decodes a bundle from a reader.
 func Read(trustDomain spiffeid.TrustDomain, r io.Reader) (*Bundle, error) {
-	panic("not implemented")
+	var b bytes.Buffer
+	_, err := b.ReadFrom(r)
```

Comment:
nit: `ioutil.ReadAll` feels more ergonomic since we just want to get at the bytes of the bundle anyway (i.e. we don't need the buffer for other reasons).